hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/HoodieCompactor.java

```diff
@@ -290,23 +291,23 @@ HoodieCompactionPlan generateCompactionPlan(
     LOG.info("Compaction looking for files to compact in " + partitionPaths + " partitions");
     context.setJobStatus(this.getClass().getSimpleName(), "Looking for files to compact: " + config.getTableName());
 
-    List<HoodieCompactionOperation> operations = context.flatMap(partitionPaths, partitionPath -> fileSystemView
-        .getLatestFileSlices(partitionPath)
-        .filter(slice -> !fgIdsInPendingCompactionAndClustering.contains(slice.getFileGroupId()))
```

Comment:
@yihua Hi, can you take a look, `fileSystemView.getLatestFileSlices` is actually not thread safe here.

Reply:
What is the usage of the `throwingFlatMapWrapper` here ?

Reply:
`fileSystemView.getLatestFileSlices` should be thread-safe.  What's the problem here?

Reply:
@danny0405 @LinMingQiang ^